core/src/main/scala/org/bitcoins/core/protocol/tlv/TLV.scala

```diff
@@ -817,6 +818,57 @@ object OracleAnnouncementV0TLV extends TLVFactory[OracleAnnouncementV0TLV] {
   }
 }
 
+sealed trait OracleAttestmentTLV extends TLV {
+  def eventId: NormalizedString
+  def publicKey: SchnorrPublicKey
+  def sigs: Vector[SchnorrDigitalSignature]
+  def outcomes: Vector[NormalizedString]
+}
+
+object OracleAttestmentTLV extends TLVParentFactory[OracleAttestmentTLV] {
+
+  val allFactories: Vector[TLVFactory[OracleAttestmentTLV]] =
+    Vector(OracleAttestmentV0TLV)
+
+  override def typeName: String = "OracleAttestmentTLV"
+}
+
+case class OracleAttestmentV0TLV(
+    eventId: NormalizedString,
+    publicKey: SchnorrPublicKey,
+    sigs: Vector[SchnorrDigitalSignature],
+    outcomes: Vector[NormalizedString])
+    extends OracleAttestmentTLV {
+  require(sigs.nonEmpty, "Cannot have 0 signatures")
+  require(
+    outcomes.size == sigs.size,
+    s"Number of outcomes must match number of signatures, ${outcomes.size} != ${sigs.size}")
+  override val tpe: BigSizeUInt = OracleAttestmentV0TLV.tpe
+
+  override val value: ByteVector = {
+    strBytes(eventId) ++
+      publicKey.bytes ++
+      u16PrefixedList(sigs) ++
+      u16PrefixedList(outcomes, strBytes)
+  }
+}
+
+object OracleAttestmentV0TLV extends TLVFactory[OracleAttestmentV0TLV] {
+  override val tpe: BigSizeUInt = BigSizeUInt(55400)
+
+  override def fromTLVValue(value: ByteVector): OracleAttestmentV0TLV = {
+    val iter = ValueIterator(value)
+
+    val eventId = iter.takeString()
+    val pubKey = SchnorrPublicKey(iter.take(32))
```

Comment:
super nit: I personally think `iter.take(SchnorrPublicKey, 32)` is more idiomatic